Automattic/jetpack
# State: Introduce state for site purchases

This PR introduces state for site purchases. This is necessary as part of the effort to load the site purchases to Jetpack. This uses an endpoint that @enejb is working on in another PR.

This can later be used (from the new selector `getSitePurchases`) in the Jetpack Backups UI to get the current purchases of the user, and will inform the UI whether the user has purchased a backup product or not.

#### Changes proposed in this Pull Request:
* Introduce site purchases state - actions, reducer and selectors.

#### Is this a new feature or does it add/remove features to an existing part of Jetpack?
* If you're an Automattician, include a shortlink to the p2 discussion with Jetpack Product here.

#### Testing instructions:
* Checkout this branch.
* Build Jetpack.
* Verify that when loading the dashboard page, you can see a request made to `purchases`
* To fully test this, the new endpoint PR by @enejb will be necessary.

#### Proposed changelog entry for your changes:
* State: Introduce state for site purchases
